### PR TITLE
feat(update): update response api get single component

### PR DESCRIPTION
**Pull request by tuannn2**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

Issue: #1978

### How To Test?

- Run command line: 
```
$ curl 'https://sw360.org/api/components/17653524' -i -X GET \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY4NTYyMDI0MCwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6ImZUeUV5bktKR19NZE9FMWF1WExWbnBFSkg1VSIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.KVoo7vqIae2M1vDp-Gxi4rFbco6uThR1c1ApDgwoN6b-QZumyGIE1AKdcExRUdQthhk2NpCprTYSqZ6uQOVfi5CLHgag45KOmIQXZdpM4OqtDzWbD8gDZDRqsAvI51syrKWNHHLm13gf9BH3XvlJ5L_Q94x5Yk-JnWv8qSgMYRDUxeGu3mDoW-ztLL5Hvp4u60xSPlOdHqI5-a9vC45X-Kr7BqSh92Z8V8dVRuGFtmWvCnI1jfuV2AcMUHxw-76_qtzq_nwoETCJ-ivMsqqRnPyrCV1KOBKNzXcgB6AlXE-NGTAXHkDwTKvLG9r2HF4aCnprdzfYDq2lm73HGvFWEA' \
    -H 'Accept: application/hal+json'
```
- Expected Output:
```
{
    "id": "17653524",
    "name": ".1",
    "description": "222",
    "createdOn": "2023-06-05",
    "componentType": "OSS",
    "subscribers": [],
    "componentOwner": "admin3@sw360.org",
    "ownerAccountingUnit": "222",
    "ownerGroup": "323",
    "ownerCountry": "AG",
    "roles": {
        "Committer": [
            "hu@gmail.com"
        ]
    },
    "visbility": "EVERYONE",
    "businessUnit": "DEPARTMENT",
    "externalIds": {
        "2": "22"
    },
    "additionalData": {
        "3": "21"
    },
    "mainLicenseIds": [],
    "defaultVendorId": "537b14af87fe4ef9b0fcf1cabe24d19f",
    "categories": [
        "11"
    ],
    "languages": [
        "Java"
    ],
    "softwarePlatforms": [
        "22"
    ],
    "operatingSystems": [
        "Microsoft Windows"
    ],
    "homepage": "http://localhost:8080",
    "mailinglist": "tuan2@gmail.com",
    "wiki": "http://localhost:8080",
    "blog": "http://localhost:8080",
    "modifiedBy": "admin@sw360.org",
    "modifiedOn": "2023-06-05",
    "setVisbility": true,
    "setBusinessUnit": true,
    "_links": {
        "self": {
            "href": "http://localhost:8080/resource/api/components/17653524"
        }
    },
    "_embedded": {
        "createdBy": {
            "email": "admin@sw360.org",
            "deactivated": false,
            "fullName": "Test Admin",
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid/ecf22aba21674e37ae2f76393dd23e9b"
                }
            }
        },
        "sw360:moderators": [
            {
                "email": "admin3@sw360.org",
                "deactivated": false,
                "fullName": "11 3",
                "_links": {
                    "self": {
                        "href": "http://localhost:8080/resource/api/users/byid/8791b4bd21374011b042b05b57409cd3"
                    }
                }
            }
        ],
        "sw360:vendors": [
            {
                "fullName": "ss",
                "_links": {
                    "self": {
                        "href": "http://localhost:8080/resource/api/vendors/e033676cb6134d1c91490472e9b36608"
                    }
                }
            }
        ],
        "sw360:releases": [
            {
                "name": ".1",
                "version": "11",
                "_links": {
                    "self": {
                        "href": "http://localhost:8080/resource/api/releases/86ebeb62c67e472ebaecab88de55c947"
                    }
                }
            }
        ],
        "defaultVendor": {
            "url": "http://localhost:8080/2ww",
            "shortName": "sssssssssss",
            "fullName": "ssssssss",
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/vendors/537b14af87fe4ef9b0fcf1cabe24d19f"
                }
            }
        },
        "modifiedBy": {
            "email": "admin@sw360.org",
            "deactivated": false,
            "fullName": "Test Admin",
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid/ecf22aba21674e37ae2f76393dd23e9b"
                }
            }
        },
        "componentOwner": {
            "email": "admin3@sw360.org",
            "deactivated": false,
            "fullName": "11 3",
            "_links": {
                "self": {
                    "href": "http://localhost:8080/resource/api/users/byid/8791b4bd21374011b042b05b57409cd3"
                }
            }
        }
    }
}
```

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Testing was successful.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java [hunk 1]
```diff
@@ -941,4 +941,22 @@ public void addEmbeddedModerationRequest(HalResource halResource, ModerationRequ
...
+                addEmbeddedUser(halResource, componentModify, "modifiedBy");
+        }
+    }
+
+    public void addEmbeddedComponentOwnerToRelease(HalResource halResource, Component sw360Component) {
+        if (sw360Component.getComponentOwner() != null) {
+            User componentOwner = getUserByEmail(sw360Component.getComponentOwner());
+            if (null != componentOwner) {
+                addEmbeddedUser(halResource, componentOwner, "componentOwner");
+                sw360Component.setComponentOwner(null);
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:958]: What's the point of line number 958 when you already checking null condition on line 956?

File: rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ComponentSpecTest.java [hunk 2]
```diff
@@ -946,6 +973,12 @@ private RestDocumentationResultHandler documentComponentProperties() {
                         fieldWithPath("ownerCountry").description("The owner country of the component"),
                         subsectionWithPath("externalIds").description("When projects are imported from other tools, the external ids can be stored here. Store as 'Single String' when single value, or 'Array of String' when multi-values"),
                         subsectionWithPath("additionalData").description("A place to store additional data used by external tools"),
+                        fieldWithPath("modifiedOn").description("The date the component was modified"),
+                        fieldWithPath("softwarePlatforms").description("The Software Platforms of componnet"),
+                        fieldWithPath("subscribers").description("The subscribers of componnet"),
+                        fieldWithPath("mainLicenseIds").description("The Main License Ids of componnet"),
+                        fieldWithPath("wiki").description("The wiki of componnet"),
+                        fieldWithPath("blog").description("The blog of componnet"),
```
Comment (rudra-superrr) [inline rest/resource-server/src/test/java/org/eclipse/sw360/rest/resourceserver/restdocs/ComponentSpecTest.java:981]: Please correct the component spelling in the above lines wherever component is used in description.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java (hunk 1 above)
Comment (tuannn2) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java:963]: The Data ComponentOwner has been added to the Embedded section so it doesn't need to be set in the Component Response.

**Comment by rudra-superrr** [comment]
Code looks good.

**Pull request merged**